testing/test_modimport.py

```diff
@@ -0,0 +1,25 @@
+import py
+import subprocess
+import sys
+import pytest
+import _pytest
+
+MODSET = [
+    x for x in py.path.local(_pytest.__file__).dirpath().visit('*.py')
+    if x.purebasename != '__init__'
+]
+
+
+@pytest.mark.parametrize('modfile', MODSET, ids=lambda x: x.purebasename)
+def test_fileimport(modfile):
+    # this test ensures all internal packages can import
+    # without needing the pytest namespace being set
+    # this is critical for the initialization of xdist
+
+    res = subprocess.call([
+        sys.executable,
+        '-c', 'import sys, py; py.path.local(sys.argv[1]).pyimport()',
+        modfile.strpath,
+    ])
+    if res:
+        pytest.fail("command result %s" % res)
```

Comment:
We should add a test that ensures the official API works:

```python
def test_pytest_api():
    import pytest
    pytest.Module
    pytest.Item
    ...
```

This will ensure we are not breaking anything now or in the future (it is easy for someone to remove an unused import somewhere and break plugins by accident).